src/main/java/com/sap/sgs/phosphor/fosstars/model/score/oss/OwaspDependencyScanScore.java

```diff
@@ -0,0 +1,127 @@
+import static com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckUsage.OPTIONAL;
+
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.feature.OwaspDependencyCheckCvssThreshold;
+import com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures;
+import com.sap.sgs.phosphor.fosstars.model.qa.ScoreVerification;
+import com.sap.sgs.phosphor.fosstars.model.qa.TestVectors;
+import com.sap.sgs.phosphor.fosstars.model.score.FeatureBasedScore;
+import com.sap.sgs.phosphor.fosstars.model.value.CVSS;
+import com.sap.sgs.phosphor.fosstars.model.value.OwaspDependencyCheckUsage;
+import com.sap.sgs.phosphor.fosstars.model.value.ScoreValue;
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.commons.lang3.Range;
+
+/**
+ * The scores assesses how well an open-source project uses OWASP Dependency Check to scan
+ * dependencies for known vulnerabilities. It is based on the following features:
+ * <ul>
+ *   <li>{@link OssFeatures#OWASP_DEPENDENCY_CHECK_USAGE}</li>
+ *   <li>{@link OssFeatures#OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD}</li>
+ * </ul>
+ */
+public class OwaspDependencyScanScore extends FeatureBasedScore {
+
+  /**
+   * The basic step score that is alloted in each step. 
+   */
+  private static final double STEP_SCORE = 3.0;
+
+  /**
+   * Initializes a new {@link OwaspDependencyScanScore}.
+   */
+  OwaspDependencyScanScore() {
+    super("How a project uses Owasp Dependency Check to scan dependencies for vulnerabilities",
+        OWASP_DEPENDENCY_CHECK_USAGE,
+        OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD);
+  }
+
+  @Override
+  public ScoreValue calculate(Value... values) {
+    Value<OwaspDependencyCheckUsage> usageValue =
+        find(OWASP_DEPENDENCY_CHECK_USAGE, values);
+    Value<Double> cvssScore =
+        find(OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD, values);
+
+    ScoreValue scoreValue = scoreValue(MIN, usageValue, cvssScore);
+
+    // if the project does not use OWASP Dependency Check
+    OwaspDependencyCheckUsage usage = usageValue.orElse(NOT_USED);
+    if (usage.equals(NOT_USED)) {
+      return scoreValue;
+    }
+
+    // if the project uses OWASP Dependency scans to identify vulnerable dependencies, then we are
+    // happy.
+    if (usage.equals(MANDATORY)) {
+      scoreValue.set(MAX - STEP_SCORE);
+    } else if (usage.equals(OPTIONAL)) {
+      scoreValue.set(STEP_SCORE);
+    }
+
+    // if the project fails build based on certain CVSS threshold. If CVSS of a vulnerability is
+    // greater than threshold, then the build fails. Lower the threshold score means most
+    // vulnerabilities are considered, then we are happy.
+    double failThreshold = cvssScore.orElse(OwaspDependencyCheckCvssThreshold.OUT_OF_BOUND);
+    if (failThreshold == CVSS.MIN) {
+      scoreValue.increase(STEP_SCORE);
+    } else if (inBetween(failThreshold)) {
+      scoreValue.increase(CVSS.MAX / (STEP_SCORE * failThreshold));
+    }
```

Comment:
How about just using a linear function for the threshold?

```java
if (CVSS.isValid(failThreshold)) {
  scoveValue.increase(STEP_SCORE * (CVSS.MAX - failThreshold) / CVSS.MAX);
}
```

Or, if you add `OwaspDependencyCheckCvssThresholdValue`:

```java
if (cvssScore.isSpecified()) {
  scoveValue.increase(STEP_SCORE * (CVSS.MAX - cvssScore.get()) / CVSS.MAX);
}
```